lib/Service/ApiService.php

```diff
@@ -67,4 +81,35 @@ public function sendFile(int $fileId): void {
 			],
 		);
 	}
+
+	public function searchMessages(string $userId, string $term, int $offset = 0, int $limit = 10): array {
+		$arguments = [
+			'format' => 'json',
+			'query' => '*' . $term . '*' ,
+		];
+
+		$paperlessURL = rtrim($this->config->url, '/') . '/api/documents/?' . http_build_query($arguments);
+		$result = $this->client->get($paperlessURL,
+			[
+				'headers' => array_merge(
+					$this->getAuthorizationHeaders(),
+					[
+						'Content-Type' => 'application/json',
+						'Accept' => 'application/json'
+					]
+				)
+			]
+		);
+
+		$body = $result->getBody();
+		$json_body = json_decode($body, true);
+
+		if (isset($json_body['error'])) {
+			return (array)$json_body;
+		}
+
+		// Sort by most recent
+		// $messages = array_reverse($json_body['document'] ?? []);
+		return array_slice($json_body, $offset, $limit);
```

Comment:
This will not work as the body will have the individual results in the `results` field.
Please also do not do the offset and limit in this service and put it in the SearchProvider.